libsel4/include/interfaces/object-api.xml

```diff
@@ -811,7 +811,7 @@
              <description>
                 An invocation for setting the Thread Local Storage (TLS) base address. This ensures that across all platforms, the TLSBase register is viewed as being completely mutable, just like all of the general purpose registers, even on platforms where modification is a privileged operation.
              </description>
-             <param dir="in" name="tls_base" type="seL4_Word"
+             <param dir="in" name="tls_base" type="seL4_Register"
```

Comment:
Can we maintain a list of all the invocation and syscalls that are having their interfaces changed to transfer hardware caps?
My understanding so far is that the following change:
- TCBSetTLSBase,
- All invocations that take or return a `seL4_UserContext`
- seL4_SetTLSBase

And what isn't expected to change:
- VSpace operations that use a virtual address for mapping or cache operations.
- seL4_TCB_SetIPCBuffer


(I assume that the [seL4_ARM_SMC](https://docs.sel4.systems/projects/sel4/api-doc.html#sel4_arm_smc) cap is out of scope at this stage too).

Reply:
Good idea. I tried to separate such changes into separate "ABI" commits. `seL4_ARM_SMC` and `seL4_ARM_VCPU_WriteRegs`, `seL4_ARM_VCPU_ReadRegs` and future RISC-V hypervisor syscalls, etc are out of scope of this PR, but will need to change when we support hypervisors. Otherwise, what you mentioned are the minimum ones that were required to support passing purecap sel4test with the default configs.